File: tree traversal/traversal.py
# python program for tree traversal
# A class that represents an individual node in a Binary tree


class Node:
    def __init__(self, key):
        self.left = None
        self.right = None
        self.val = key

# A function to do inorder tree traversal


def printInorder(root):
    if root:
        # first recur on left child
        printInorder(root.left)
        print(root.val)
        printInorder(root.right)

# A function to do postorder tree traversal


def printPostorder(root):
    if root:
        printPostorder(root.left)
        printPostorder(root.right)
        print(root.val)

# A function to do Preorder tree traversal


def printPreorder(root):
    if root:
        print(root.val)
        printPreorder(root.left)
        printPreorder(root.right)


# Driver code
root = Node(1)
root.left = Node(2)
root.right = Node(3)
root.left.left = Node(4)
root.left.right = Node(5)

print("Preorder traversal: ")
printPreorder(root)

print("Inoder traversal: ")
printInorder(root)

print("postorder traversal: ")
printPostorder(root)

File: tree traversal/traversal.java
// java program for different tree traversal
//  class containing left and right chlld of current Node
// and key value
class Node {
	int key;
	Node left, right;
	public Node(int item)
	{
		key = item;
		left = right = null;
	}
}

class Traversal
{
	// Root of Binary tree
	Node root;
	Traversal()
	{
		root = null;
	}


	// Given a binary tree, print its nodes according to the "bottom-up"
	// post order traversal
	void printPostorder(Node node)
	{
		if (node == null) {
			return;
		}
		printPostorder(node.left);
		printPostorder(node.right);
		System.out.print(node.key + " ");
	}


	/* Given a binary tree, print its nodes in inorder*/
	void printInorder(Node node)
	{
		if (node == null) {
			return;
		}
		printInorder(node.left);
		System.out.print(node.key + " ");
		printInorder(node.right);
	}


	/* Given a binary tree, print its nodes in preorder*/
	void printPreorder(Node node)
	{
		if (node == null) {
			return;
		}
		System.out.print(node.key + " ");
		printPreorder(node.left);
		printPreorder(node.right);
	}


	// Wrapper over the above recursive functions
	void printPostorder() { printPostorder(root); }
	void printInorder() { printInorder(root); }
	void printPreorder() { printPreorder(root); }

	// Driver method
	public static void main(String[] args)
	{
		Traversal tree = new Traversal();

		tree.root = new Node(1);
		tree.root.left = new Node(2);
		tree.root.right = new Node(3);
		tree.root.left.left = new Node(4);
		tree.root.left.right = new Node(5);

		System.out.println("Preorder traversal of binary tree is ");
		tree.printPreorder();

		System.out.println("\nInorder traversal of binary tree is ");
		tree.printInorder();

		System.out.println("\nPostorder traversal of binary tree is ");
		tree.printPostorder();
	}
}
